src/main/java/dinkplugin/notifiers/ExternalPluginNotifier.java

```diff
@@ -87,7 +88,7 @@ private void handleNotify(ExternalNotificationRequest input) {
         var player = Utils.getPlayerName(client);
         var template = Template.builder()
             .template(input.getText())
-            .replacements(input.getReplacements())
+            .replacements(Objects.requireNonNullElse(input.getReplacements(), Map.of()))
```

Comment:
```suggestion
            .replacements(Objects.requireNonNullElse(input.getReplacements(), new HashMap<>(2)))
```

tiny nitpick; `Map.of` yields an immutable collection which could be problematic for adding the `%USERNAME%` entry. in reality, lombok has a current implementation detail where maps marked as `@Singular` are actually represented by a list of keys & list of values internally (and a new map is created upon `.build()`) so the immutability doesn't bite us today